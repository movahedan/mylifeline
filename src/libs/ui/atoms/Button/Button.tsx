import { classNames } from "@mylifeline/utilities";

import type { CSSProperties, ReactNode } from "react";

type ButtonProps = {
  icon?: ReactNode;
  dataTestId?: string;
  style?: CSSProperties;
  className?: string;
  children: ReactNode;
};

export const Button = ({
  icon,
  dataTestId,
  style,
  className,
  children,
}: ButtonProps) => (
  <button
    data-testid={dataTestId}
    style={style}
    className={classNames([
      "inline-flex prose prose-button py-16 px-40",
      "border rounded-lg outline-0",
      "transition-all duration-300",
      className || "",
    ])}
  >
    {icon && <span className="mr-20">{icon}</span>}
    {children}
  </button>
);
