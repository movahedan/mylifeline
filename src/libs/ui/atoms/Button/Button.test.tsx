import { render } from "@testing-library/react";

import { Button } from "./Button";

describe("<Button>", () => {
  it("should match is's snapshot", () => {
    const { container, rerender } = render(<Button>Hello</Button>);
    expect(container).toMatchSnapshot("default");

    rerender(<Button icon="i">Hello</Button>);
    expect(container).toMatchSnapshot("icon default");
  });
});
