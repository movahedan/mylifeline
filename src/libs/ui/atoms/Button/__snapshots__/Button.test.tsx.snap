// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button> should match is's snapshot: default 1`] = `
<div>
  <button
    class="inline-flex prose prose-button py-16 px-40 border rounded-lg outline-0 transition-all duration-300"
  >
    Hello
  </button>
</div>
`;

exports[`<Button> should match is's snapshot: icon default 1`] = `
<div>
  <button
    class="inline-flex prose prose-button py-16 px-40 border rounded-lg outline-0 transition-all duration-300"
  >
    <span
      class="mr-20"
    >
      i
    </span>
    Hello
  </button>
</div>
`;
